rust/lance-file/src/v2/writer.rs

```diff
@@ -185,8 +185,14 @@ impl FileWriter {
         //
         // Also, there is no point in trying to make write_page parallel anyways
         // because we wouldn't want buffers getting mixed up across pages.
+        let mut encoded_pages = Vec::new();
         while let Some(encoding_task) = encoding_tasks.next().await {
             let encoded_page = encoding_task?;
+            //self.write_page(encoded_page).await?;
+            encoded_pages.push(encoded_page);
+        }
+        encoded_pages.sort_by_key(|p| p.page_idx);
+        for encoded_page in encoded_pages.into_iter() {
             self.write_page(encoded_page).await?;
```

Comment:
this have performance implications, we are waiting compute then do IO, I want to have a better way to `batch size independent` + `decouple compute/IO`